projects/plugins/crm/includes/jpcrm-templating-placeholders.php

```diff
@@ -1654,7 +1654,8 @@ public function replace_placeholders(
 		$string = '',
 		$replacements = array(),
 		$replacement_objects = false,
-		$retain_unset_placeholders = false
+		$retain_unset_placeholders = false,
+		$include_non_rendered_quote_keys = true
```

Comment:
I'd rather avoid object-specific params for a generic function like this. Three alternatives to consider:

1. Somehow leverage the `$retain_unset_placeholders` to perform the same function. At a glance, it seems the purpose of this param is to not render unset placeholders, which seems to have a fair amount of overlap.
2. Generify `$include_non_rendered_quote_keys` to instead accept an array of keys to retain.
3. In a combo of 1 and 2, allow `$retain_unset_placeholders` to accept either a bool or an array of keys to retain.

If you're using the same keys over and over, you could add a wrapper function that passes the appropriate keys.

Reply:
Thanks, yes, that makes sense. I went with option 2 here in another commit: 1f87ec0fcb9204136def853cddbcbf9be642f5b3